### PR TITLE
Allow to install the library as a CMake project

### PR DESCRIPTION
**Pull request by CharlesGueunet**
(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
---------------------------------------------------

These changes add the cmake mechanism to install the project so it can be used in a find_package by other projects.

Any relevant logs, error output, etc?
-------------------------------------
There is one change in this MR which is just commenting some lines causing a linker error on recent clang compilers.  The error is provided in comment

Where has this been tested?
---------------------------
**Operating System:** Gentoo Linux

I also used fespp to ensure this library is correctly installed.

Other comments:
---------------

This PR is related to another MR updating fespp to uses this library as expected (will be provided here when available).

### CODE REVIEW

**Comment by philippeVerney** [comment]
Thanks a lot @CharlesGueunet 
I try to look at it deeper next week and wil let you know.

**Comment by philippeVerney** [comment]
@CharlesGueunet Would it be possible to target the dev master instead of the master one?

**Comment by CharlesGueunet** [comment]
> @CharlesGueunet Would it be possible to target the dev branch instead of the master one?

Done 

> I try to look at it deeper next week and wil let you know.

Thank you

**Review by philippeVerney — CHANGES_REQUESTED** [review]
Thanks for all this work.
Not sure to have enough CMAKE knowledge to deeply understand the proposed changes but hopefully my review should be a bit useful.
I have a commit to push/patch right after this review.

**Comment by philippeVerney** [comment]
Here is a patch trying to resolve my comments above.
[0001-Proposal-after-review.patch](https://github.com/F2I-Consulting/fesapi/files/14670410/0001-Proposal-after-review.patch)

I also rebased on origin/dev to check and it is fine.

File: src/FesapiCppConfig.cmake.in
[the remark this answers is not in the document]
Comment (CharlesGueunet) [reply src/FesapiCppConfig.cmake.in:1]: I have slightly updated your changes so the .in is in the cmake directory, but the generated file stay in the build dir only (and not in the source).

**Pull request merged**